2-Browsers/Week1/homework/ex5-catWalk/index.js

```diff
@@ -15,8 +15,35 @@
    for 5 seconds, and then replace the img with the original image and have it 
    continue the walk.
 -----------------------------------------------------------------------------*/
+
+const catImage = document.querySelector('img');
+catImage.style.left = '0px';
+const catImageWidth = catImage.width;
```

Comment:
If you add a `console.log(catImageWidth)` here you'll note that it's zero! That's because you're calculating the width before the image has fully loaded. You'll need to add an `catImage.addEventListener('load', () => ...)` to get the image's width _after_ the image has loaded.

Then you'll have to do the rest of your calculations after you have that width.

Reply:
Actually , I didn't understand where and why do I have to put addEventListener to calculate the image width when loading the page;

I thought that the image is already known and declared when I appended it with a variable in JS using a document.querySelector('img') at the beginning of the code
 const catImage = document.querySelector('img');
 
 so catImage is the image with all properties 


can you explained a little bit please

and thank you very much in advanced

Reply:
The image has to be downloaded from its `src` before the browser knows the width of the image. Because it can take a long time to download images, your browser will do that in the background while it continues running your Javascript. so your Javascript is being executed _before_ the image is downloaded, and so `catImage.width` will be zero, because it can't know the image's width after it's been fully downloaded.

Reply:
Sorry for late reply,
Ok then in this case is the using of defer will solve this issue,
 then I'll be sure that the script will not run until after the page has loaded.
 because the image is already in HTML structure
 
 
 <script src="index.js" defer></script>
 
 is that correct 😄